### ISSUE TITLE
block-scoped-var problem with switch statement invoked in a function

### ISSUE THREAD

**Issue by chestnutchen** [eslint/eslint#1453]
With the following code:

``` javascript
    /* eslint block-scoped-var: 2, no-undef: 0, strict: 0 */
    function test(a) {
        return a + ", but it warns";
    }

    function testSwitch (str) {
        switch (str) {
            case "test":
                var a = "should not warn";
                var b = test(a);
                /* eslint-disable no-alert, no-console */
                console.log(b);
                /* eslint-enable no-alert, no-console */
                break;

            default:
                break;
        }
    }

    testSwitch("test");
```

run eslint and I got:

```
10:29  error  "a" used outside of binding context        block-scoped-var
12:16  error  "console" used outside of binding context  block-scoped-var
12:28  error  "b" used outside of binding context        block-scoped-var
```
